### ISSUE TITLE
rails 6.1 / ruby 2.7.2

### ISSUE THREAD

**Issue by michelson** [bobf/action_mailbox_amazon_ingress#4]
Hi , I've tried to run the gem specs and specs fails

```
➜  action_mailbox_amazon_ingress git:(master) ✗ bundle exec rspec
== 20200508151055 CreateActiveStorageTables: migrating ========================
-- create_table(:active_storage_blobs, {})
   -> 0.0014s
-- create_table(:active_storage_attachments, {})
   -> 0.0011s
== 20200508151055 CreateActiveStorageTables: migrated (0.0026s) ===============

== 20200508151056 CreateActionMailboxTables: migrating ========================
-- create_table(:action_mailbox_inbound_emails, {})
   -> 0.0018s
== 20200508151056 CreateActionMailboxTables: migrated (0.0019s) ===============


ActionMailboxAmazonIngress::RSpec::Email
  is expected to be a kind of ActionMailboxAmazonIngress::RSpec::Email
  url
    is expected to eql "/rails/action_mailbox/amazon/inbound_emails"
  headers
    is expected to eql {"content-type"=>"application/json"}
  params
    is expected to eql {"Message"=>"{\"notificationType\":\"Received\",\"content\":\"raw encoded email\"}", "TopicArn"=>"topic:arn:default", "Type"=>"Notification"}
  not authentic
    authentic?
      is expected to eql false
  authentic
    authentic?
      is expected to eql true
  custom topic
    params
      is expected to include {"TopicArn" => "custom-topic"}

ActionMailboxAmazonIngress::RSpec::SubscriptionConfirmation
  is expected to be a kind of ActionMailboxAmazonIngress::RSpec::SubscriptionConfirmation
  url
    is expected to eql "/rails/action_mailbox/amazon/inbound_emails"
  headers
    is expected to eql {"content-type"=>"application/json"}
  params
    is expected to eql {"SubscribeURL"=>"http://example.com/subscribe", "TopicArn"=>"topic:arn:default", "Type"=>"SubscriptionConfirmation"}
  not authentic
    authentic?
      is expected to eql false
  authentic
    authentic?
      is expected to eql true
  custom topic
    params
      is expected to include {"TopicArn" => "custom-topic"}

ActionMailboxAmazonIngress
  has a version number

inbound email
  receives inbound email (FAILED - 1)

rspec
  topic subscription
    recognized topic
      renders 200 OK
    unrecognized topic
      renders 401 Unauthorized
  email delivery
    recognized topic
      renders 204 No Content (FAILED - 2)
      delivers an email to inbox (FAILED - 3)
    unrecognized topic
      renders 401 Unauthorized

subscription confirmation
  valid Amazon SSL signature
    fetches subscription URL
  invalid Amazon SSL signature
    does not fetch subscription URL
  unrecognized topic
    does not fetch subscription URL
  recognized topic
    fetches subscription URL

Failures:

  1) inbound email receives inbound email
     Failure/Error: expect(ActionMailbox::InboundEmail.count).to eql 1
     
       expected: 1
            got: 0
     
       (compared using eql?)
     # ./spec/requests/inbound_email_spec.rb:13:in `block (2 levels) in <top (required)>'
     # /Users/michelson/.rvm/gems/ruby-2.7.1/gems/webmock-3.12.2/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

  2) rspec email delivery recognized topic renders 204 No Content
     Failure/Error: expect(response).to have_http_status :no_content
       expected the response to have status code :no_content (204) but it was :internal_server_error (500)
     # ./spec/requests/rspec_spec.rb:33:in `block (4 levels) in <top (required)>'
     # /Users/michelson/.rvm/gems/ruby-2.7.1/gems/webmock-3.12.2/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

  3) rspec email delivery recognized topic delivers an email to inbox
     Failure/Error: expect(ActionMailbox::InboundEmail.last.mail.recipients).to eql ['user@example.com']
     
     NoMethodError:
       undefined method `mail' for nil:NilClass
     # ./spec/requests/rspec_spec.rb:38:in `block (4 levels) in <top (required)>'
     # /Users/michelson/.rvm/gems/ruby-2.7.1/gems/webmock-3.12.2/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

Finished in 0.28573 seconds (files took 3.9 seconds to load)
25 examples, 3 failures

Failed examples:

rspec ./spec/requests/inbound_email_spec.rb:10 # inbound email receives inbound email
rspec ./spec/requests/rspec_spec.rb:31 # rspec email delivery recognized topic renders 204 No Content
rspec ./spec/requests/rspec_spec.rb:36 # rspec email delivery recognized topic delivers an email to inbox

➜  action_mailbox_amazon_ingress git:(master) ✗ 
```

my diff:

```
diff --git a/.ruby-version b/.ruby-version
index aedc15b..37c2961 100644
--- a/.ruby-version
+++ b/.ruby-version
@@ -1 +1 @@
-2.5.3
+2.7.2
diff --git a/action_mailbox_amazon_ingress.gemspec b/action_mailbox_amazon_ingress.gemspec
index 1e2dc85..236b387 100644
--- a/action_mailbox_amazon_ingress.gemspec
+++ b/action_mailbox_amazon_ingress.gemspec
@@ -26,7 +26,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
 
   spec.add_runtime_dependency 'aws-sdk-sns', '~> 1.23'
-  spec.add_dependency 'rails', '~> 6.0'
+  spec.add_dependency 'rails', '~> 6.1'
 
   spec.add_development_dependency 'betterp', '~> 0.1.3'
   spec.add_development_dependency 'devpack', '~> 0.2.1'
```

**Comment by bobf** [comment]
@michelson Thanks for reporting. I have locked to Rails `< 6.1` for now - I will try to fix these issues ASAP and then we can remove the constraint. PRs are also welcome. : )

For ref:

```
commit 5da41a1874fb3cffe6577a751e4e526c2a3773d0
Author: Bob Farrell <git@bob.frl>
Date:   Mon Mar 29 21:37:08 2021 +0100

    Lock to Rails < 6.1
    
    Tests do not pass with Rails 6.1 - locking to < 6.1 until issues are
    resolved.
    
    https://github.com/bobf/action_mailbox_amazon_ingress/issues/4

diff --git a/action_mailbox_amazon_ingress.gemspec b/action_mailbox_amazon_ingress.gemspec
index 1e2dc85..2441c33 100644
--- a/action_mailbox_amazon_ingress.gemspec
+++ b/action_mailbox_amazon_ingress.gemspec
@@ -26,7 +26,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
 
   spec.add_runtime_dependency 'aws-sdk-sns', '~> 1.23'
-  spec.add_dependency 'rails', '~> 6.0'
+  spec.add_dependency 'rails', '~> 6.0', '< 6.1'
 
   spec.add_development_dependency 'betterp', '~> 0.1.3'
   spec.add_development_dependency 'devpack', '~> 0.2.1'
```

**Comment by michelson** [comment]
Hi @bobf , Interesting, I can confirm that tests are green for < 6.1 , 
I could try a PR , but I don't know really what changed on rails , any hints?

**Comment by bobf** [comment]
@michelson I'm afraid not ! The first thing I would do is dig into the Rails changelog and maybe ActionMailbox code to see what's going on, but other than that you probably know about as much as I do about it. : )

I'm happy to fix it myself, just might not be for a few days or so - I use this lib at work and I would definitely like to upgrade to 6.1, so I have that incentive on top of just simply wanting my code to be useful to people. If you can figure out a solution then I'd be happy to accept the PR but don't stress if not, I'll get it sorted soon-ish.

**Comment by bobf** [comment]
@michelson This should work for you now - Rails 6.1 ActiveStorage has a couple of extra DB migrations that were needed. I generated them with `rails active_storage:update` and then re-ran the tests (the tests run migrations automatically) and everything magically worked. : )

Let me know if you have any more problems and thanks for taking the time to report the issue.

BTW I detected the problem by tailing logs - `spec/support/dummy/log/production.log`. There was an error in the logs which pointed me to [this StackOverflow answer](https://stackoverflow.com/a/61442407/12658305).
